RoddyCore/test/de/dkfz/roddy/tools/ComplexLineTest.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2017 eilslabs.
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+
+package de.dkfz.roddy.tools
+
+import groovy.transform.CompileStatic
+import org.junit.Test
+
+/**
+ * A test class for the Complex Line class
+ * Created by heinold on 16.01.17.
+ */
+@CompileStatic
+class ComplexLineTest {
+
+
+    public static final String commandLineExample = "--abc --def=\"ab,def;efk=test;anotherValue='equals'\" --jki --a=\"something=def\""
+    public static final String[] splittedCommandLineExample = [
+            "--abc",
+            "--def=\"ab,def;efk=test;anotherValue='equals'\"",
+            "--jki",
+            "--a=\"something=def\""
+    ]
+
+    public static final String codeLineExample = "def multiline='a string', b={something[in a braquet]}, method(a(b()))"
+    public static final String[] splittedCodeLineExample = [
+            "def multiline='a string'",
+            " b={something[in a braquet]}",
+            " method(a(b()))"
+    ]
+
+    public static final String faultyLineExample = "a(b(c())"
```

Comment:
I'd leave it for now ok? I'll put some comments into the test class and see, how everything works out for the current cases.